crates/sui-core/src/authority.rs

```diff
@@ -1029,10 +1050,19 @@ impl AuthorityState {
                 epoch_store.epoch(),
             );
 
+        let effects_with_events = effects.clone();
+
+        if epoch_store.epoch() >= REMOVE_EFFECTS_EPOCH {
```

Comment:
I think here we should also check if this is a validator

Reply:
Doing so would cause this error to trigger: https://github.com/MystenLabs/sui/blob/testnet/crates/sui-core/src/authority.rs#L636.

As far as I can tell right now, fullnode only serves events from the event store, which is unaffected by this change. We should be fine to remove events from signed effects both for validators and full nodes.